hack/generate-cd.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+SOURCE_PATH="$(dirname $0)/.."
+IMAGE_REGISTRY="$(${SOURCE_PATH}/hack/get-image-registry.sh)"
+CD_REGISTRY="$(${SOURCE_PATH}/hack/get-cd-registry.sh)"
+
+CA_PATH="$(mktemp -d)"
+TMP_COMPONENT_DESCRIPTOR_PATH="${CA_PATH}/component-descriptor.yaml"
+
+if ! which component-cli 1>/dev/null; then
+  echo -n "component-cli is required to generate the component descriptors"
+  echo -n "Trying to installing it..."
+  go get github.com/gardener/component-cli/cmd/component-cli
```

Comment:
Move this to the `install-requirements` target in the Makefile.

Reply:
I'm not sure if we should do this because if we move the script to a central location do we know if all projects have some install-requirements target? And we have to adapt this for all other projects.